jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/wizard/SamInitSelectionPanel.kt

```diff
@@ -0,0 +1,163 @@
+import software.aws.toolkits.jetbrains.services.lambda.RuntimeGroup.Companion.find
+import software.aws.toolkits.jetbrains.services.lambda.runtimeGroup
+import software.aws.toolkits.jetbrains.services.lambda.sam.SamExecutable
+import java.awt.BorderLayout
+import javax.swing.JButton
+import javax.swing.JComponent
+import javax.swing.JPanel
+import javax.swing.JTextField
+
+class SamInitSelectionPanel(
+    wizardFragmentList: List<WizardFragment>,
+    private val projectLocation: TextFieldWithBrowseButton? = null, /* Only available in PyCharm! */
+    runtimeFilter: (Runtime) -> Boolean = { true }
+) {
+    lateinit var mainPanel: JPanel
+    private lateinit var runtimeComboBox: ComboBox<Runtime>
+    private lateinit var samExecutableField: JTextField
+    private lateinit var editSamExecutableButton: JButton
+    private lateinit var samLabel: JBLabel
+    private lateinit var templateComboBox: ComboBox<SamProjectTemplate>
+    private lateinit var fragments: Wrapper
+
+    private val wizardFragments: Map<WizardFragment, JComponent>
+
+    init {
+        // Source all templates, find all the runtimes they support, then filter those by what the IDE supports
+        val supportedRuntimeGroups = RuntimeGroup.registeredRuntimeGroups()
+        SamProjectTemplate.supportedTemplates().asSequence()
+            .flatMap { it.supportedRuntimes().asSequence() }
+            .filter(runtimeFilter)
+            .filter { supportedRuntimeGroups.contains(find { runtimeGroup -> runtimeGroup.runtimes.contains(it) }) }
+            .distinct()
+            .sorted()
+            .forEach { runtimeComboBox.addItem(it) }
+
+        setupSamSelectionElements(samExecutableField, editSamExecutableButton, samLabel)
+
+        runtimeComboBox.addActionListener {
+            runtimeUpdate()
+            wizardUpdate()
+        }
+
+        templateComboBox.addActionListener { wizardUpdate() }
+        templateComboBox.renderer = SimpleListCellRenderer.create { label, value, _ ->
+            label.text = value?.displayName()
+            label.toolTipText = value?.description()
+        }
+
+        wizardFragments = wizardFragmentList.associateWith {
+            val panel = JPanel(BorderLayout())
+            val fragmentTitle = it.title()
+            if (fragmentTitle != null) {
+                panel.border = IdeBorderFactory.createTitledBorder(it.title(), false)
+            }
+            panel.add(it.component(), BorderLayout.CENTER)
+            panel
+        }
+
+        fragments.setContent(
+            panel {
+                wizardFragments.values.forEach {
+                    row() {
+                        it(grow)
+                    }
+                }
+            }
+        )
+
+        runtimeUpdate()
+        wizardUpdate()
+    }
+
+    fun setRuntime(runtime: Runtime) {
+        val itemCount = runtimeComboBox.itemCount
+        for (itemIndex in 0 until itemCount) {
```

Comment:
nit: is index iteration the best way to do this?

Reply:
Uhh, prob not. This was J2K's output.

Not sure why it was does like this in Java either, since setSelectedItem already validates its a valid item (if not editable)